### PR TITLE
Update base version to 3.3.1-RC1

### PR DESCRIPTION
**Pull request by Kordyjan**
Also fix incorrect tasty version

### CODE REVIEW

File: tasty/src/dotty/tools/tasty/TastyFormat.scala [hunk 1]
```diff
@@ -289,7 +289,7 @@ object TastyFormat {
    *  compatibility, but remains backwards compatible, with all
    *  preceeding `MinorVersion`.
    */
-  final val MinorVersion: Int = 2
+  final val MinorVersion: Int = 4
```
Comment (Kordyjan) [inline tasty/src/dotty/tools/tasty/TastyFormat.scala:292]: During the merge of `language-reference-stable`, git somehow decided there was no conflict here.
I'll make it impossible for ports from the documentation to change anything except the documentation itself and the scaladoc.

File: project/MiMaFilters.scala [hunk 2]
```diff
@@ -5,10 +5,29 @@ object MiMaFilters {
...
+    ProblemFilters.exclude[DirectMissingMethodProblem]("dotty.tools.tasty.TastyBuffer.reset"),
+    ProblemFilters.exclude[DirectMissingMethodProblem]("dotty.tools.tasty.TastyFormat.APPLYsigpoly"),
+    ProblemFilters.exclude[DirectMissingMethodProblem]("dotty.tools.tasty.TastyHash.pjwHash64"),
+    ProblemFilters.exclude[DirectMissingMethodProblem]("dotty.tools.tasty.util.Util.dble")
   )
-  val TastyCore: Seq[ProblemFilter] = Seq()
   val Interfaces: Seq[ProblemFilter] = Seq(
-    ProblemFilters.exclude[MissingClassProblem]("dotty.tools.dotc.interfaces.DiagnosticRelatedInformation"),
-    ProblemFilters.exclude[ReversedMissingMethodProblem]("dotty.tools.dotc.interfaces.Diagnostic.diagnosticRelatedInformation")
+    ProblemFilters.exclude[ReversedMissingMethodProblem]("dotty.tools.dotc.interfaces.Diagnostic.diagnosticRelatedInformation"),
```
Comment (prolativ) [inline project/MiMaFilters.scala:29]: I would expect these exclusions would disappear after the release of 3.3 - not that we would need more for previously added parts of the API

**Review by prolativ — APPROVED** [review]

File: tasty/src/dotty/tools/tasty/TastyFormat.scala (hunk 1 above)
Comment (nicolasstucki) [reply tasty/src/dotty/tools/tasty/TastyFormat.scala:292]: This should have been minor version `3`.

Comment (nicolasstucki) [reply tasty/src/dotty/tools/tasty/TastyFormat.scala:292]: Acctualy, I was wrong, this was only used to release `3.4.0-RC1`. Therefore the version is correct.

**Pull request merged**